src/hotspot/cpu/aarch64/frame_aarch64.cpp

```diff
@@ -262,9 +263,13 @@ bool frame::safe_for_sender(JavaThread *thread) {
     return false;
   }
 
-  // Will the pc we fetch be non-zero (which we'll find at the oldest frame)
+  // Will the pc we fetch be non-zero (which we'll find at the oldest frame) and readable
 
-  if ( (address) this->fp()[return_addr_offset] == NULL) return false;
+  if (!thread->is_in_usable_stack((address)this->fp())) {
```

Comment:
Is this check really necessary? 'fp_safe' is true when we come here. Then it looks to me that the two preceding checks has already ensured that condition 'thread->is_in_usable_stack((address)this->fp())' will always evaluate to true here.

[1] https://github.com/openjdk/jdk/blob/master/src/hotspot/cpu/aarch64/frame_aarch64.cpp#L68
[2] https://github.com/openjdk/jdk/blob/master/src/hotspot/cpu/aarch64/frame_aarch64.cpp#L90

Reply:
I'll get crashes in my fuzzing test case if I leave it out.

Reply:
I agree fp has already been validated.

Reply:
But has fp + return_addr_offset?

Reply:
Probably not, but not relevant to the comments about line  268.

Reply:
I removed the check.